js/ui/appActivation.js

```diff
@@ -0,0 +1,424 @@
+// -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
+
+const Clutter = imports.gi.Clutter;
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+const GObject = imports.gi.GObject;
+const Lang = imports.lang;
+const Mainloop = imports.mainloop;
+const Meta = imports.gi.Meta;
+const Shell = imports.gi.Shell;
+const Signals = imports.signals;
+const St = imports.gi.St;
+
+const Background = imports.ui.background;
+const GrabHelper = imports.ui.grabHelper;
+const Main = imports.ui.main;
+const Overview = imports.ui.overview;
+const Panel = imports.ui.panel;
+const Tweener = imports.ui.tweener;
+const Util = imports.misc.util;
+const WindowManager = imports.ui.windowManager;
+
+const SPLASH_SCREEN_TIMEOUT = 700; // ms
+
+// By default, maximized windows are 75% of the workarea
+// of the monitor they're on when unmaximized.
+const DEFAULT_MAXIMIZED_WINDOW_SIZE = 0.75;
+const LAUNCH_MAXIMIZED_DESKTOP_KEY = 'X-Endless-LaunchMaximized';
+
+
+// Determine if a splash screen should be shown for the provided
+// ShellApp and other global settings
+function _shouldShowSplash(app) {
+    let info = app.get_app_info();
+
+    if (!(info && info.has_key(LAUNCH_MAXIMIZED_DESKTOP_KEY) &&
+         info.get_boolean(LAUNCH_MAXIMIZED_DESKTOP_KEY))) {
```

Comment:
Ah, I see... I normally leave the braces if either the condition or the body spans through multiple lines (e.g. in WebKit), but I'm ok with changing it if that's the desired style here. Changed